Sources/NIOOpenSSL/SSLPrivateKey.swift

```diff
@@ -68,7 +161,13 @@ public class OpenSSLPrivateKey {
         let key: UnsafeMutablePointer<EVP_PKEY>?
         switch format {
         case .pem:
-            key = PEM_read_bio_PrivateKey(bio, nil, nil, nil)
+            if let userData = pemUserData {
+                // This annoying conditional binding is used to work around the fact that I cannot pass
+                // a variable to a function pointer argument.
+                key = PEM_read_bio_PrivateKey(bio, nil, globalOpenSSLPassphraseCallback(buf:size:rwflag:u:), userData)
+            } else {
+                key = PEM_read_bio_PrivateKey(bio, nil, nil, nil)
+            }
         case .der:
             key = d2i_PrivateKey_bio(bio, nil)
```

Comment:
should we assert this isn't happening on an event loop as this stuff will block?

Reply:
I don't think so. This blocking is potential, and not necessarily wildly egregious, so I'm inclined to leave it in the user's hands. Regardless we certainly can't do it in this patch, as the assertion would be semver major.